packages/Kokkos/src/DTK_Mesh_def.hpp

```diff
@@ -77,21 +77,22 @@ Mesh<SC, LO, GO, NO>::Mesh( const Teuchos::RCP<const Teuchos::Comm<int>> &comm,
         size_t pos = 0;
         size_t cell = 0;
         // Transform the 1D view of the connectivity to a 2D view
-        connectivity_view_2d connectivity_2d( "connectivity_2d",
-                                              n_cells_current_topo, n_nodes );
-        for ( size_t i = 0; i < topology.size(); ++i )
+        connectivity_view_2d connectivity_2d(
+            "connectivity_2d", n_cells_current_topo, n_nodes_per_cell );
+        for ( size_t i = 0; i < n_cells; ++i )
```

Comment:
I know that this code is going away, but in general, which could be a single loop where you fill multiple data arrays. This way, you only need one pass through all the data rather multiple (=number of different topologies).